logic/Rule.java

```diff
@@ -317,7 +284,11 @@ public static Conclusion create(Rule rule) {
             throw TypeDBException.of(ILLEGAL_STATE);
         }
 
-        public Conjunction conjunction() {
+        public ResolvableConjunction conjunction() {
+            return conjunction;
+        }
+
+        public Conjunction pattern() {
             return rule().then();
         }
 
```

Comment:
Try to make the ResolvableConjunction the important-member of a Rule:
Rename the method returning the pattern.Conjunction to 'pattern'
Use the `conjunction` name for the one returning a ResolvableConjunction.